flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,250 @@
+            4: 'Connection Refused, bad user name or password - The data in the user name or password is malformed',
+            5: 'Connection Refused, not authorized - The Client is not authorized to connect',
+        }
+
+        # Handle the MQTT broker protocol to be used
+        if config.get('broker_protocol') == 'MQTTv311':
+            config['broker_protocol_class'] = mqtt.MQTTv311
+        else:
+            config['broker_protocol_class'] = mqtt.MQTTv31
+
+        logger.trace('MQTT notify config={}', config)
+
+        # create the mqtt client
+        client = mqtt.Client(
+            protocol=config['broker_protocol_class'], transport=config['broker_transport']
+        )
+
+        client.enable_logger(logger=logger)
+
+        client.on_log = on_log_cb
+        client.on_publish = on_publish_cb
+        client.on_disconnect = on_disconnect_cb
+
+        # Handle SSL/TLS communication w/out certificate authentication
+        if not config.get('certificates', {}).get('client_cert') and config.get(
+            'encrypted_communication'
+        ):
+            client.tls_set(
+                ca_certs=certs.get('broker_ca_cert'),
+                certfile=None,
+                keyfile=None,
+                cert_reqs=ssl.CERT_NONE,
+            )
+
+            client.tls_insecure_set(True)
+
+            logger.verbose('Basic SSL/TLS encrypted communications enabled')
+            logger.verbose('TLS insecure cert mode enabled. Broker cert will not be validated')
+
+        # Handle SSL/TLS communication with certificate authentication
+        if config.get('certificates'):
+            certs = config['certificates']
+            logger.debug('TLS certificate config: {}', certs)
+
+            tls_version_map = {
+                'tlsv1.2': ssl.PROTOCOL_TLSv1_2,
+                'tlsv1.1': ssl.PROTOCOL_TLSv1_1,
+                'tlsv1': ssl.PROTOCOL_TLSv1,
+                '': None,
+            }
+            tls_version = tls_version_map.get(certs.get('tls_version'), ssl.PROTOCOL_TLSv1_2)
+            logger.verbose('TLS version is {}', tls_version)
+
+            cert_required = (
+                ssl.CERT_REQUIRED if certs.get('validate_broker_cert', True) else ssl.CERT_NONE
+            )
+
+            client.tls_set(
+                ca_certs=certs.get('broker_ca_cert'),
+                certfile=certs.get('client_cert'),
+                keyfile=certs.get('client_key'),
+                cert_reqs=cert_required,
+                tls_version=tls_version,
+            )
+
+            if not certs.get('validate_broker_cert'):
+                client.tls_insecure_set(True)
+                logger.debug('TLS insecure cert mode enabled. Broker cert will not be validated')
+            else:
+                logger.debug('TLS secure cert mode enabled. Broker cert will be validated')
+
+        # Handle user/pass authentication
+        if config.get('username'):
```

Comment:
When checking key presence in dict the pythonic way is `if "username" in config:`, same applies to few other places in the code.